### ISSUE TITLE
HoK doesn't work properly with Electrolysis in firefox 49+ (some versions of 48 too)

### ISSUE THREAD

**Issue by shopclerk** [mooz/keysnail#213]
Electrolysis will arrive forcibly to Firefox in a couple of versions, and this could affect some parts of the plugin as well.  May something be done to it?

https://wiki.mozilla.org/Electrolysis

**Comment by shopclerk** [comment]
This also affects some keybindings such as C-n as well.

**Comment by luka-n** [comment]
I had to disable electrolysis for this, unfortunately hok is more useful. Anyone had a look at this yet?

**Comment by tosmarcel** [comment]
I'm also on firefox 48 with electrolysis enabled and I'm experiencing the same problem.

**Comment by kccai** [comment]
Yes, I can confirm that the electrolysis (or Multi Process Feature) breaks keysnail ... Even the basic `jk` scrolling keys do not work ...

**Comment by louiz** [comment]
Confirming this as well.

* “Scroll line down” and “Scroll line up” DON’T work.
* but “Scroll page up” and “Scroll page down” do work.

Tested with firefox 49 with electrolysis enabled.

**Comment by phil-s** [comment]
I just looked up how to disable this electrolysis feature, and the option name for it is quite astonishingly unintuitive. I've not yet had to do this myself, but hopefully this helps anyone reading this thread to find out how to get KeySnail working again.

http://www.ghacks.net/2016/07/22/multi-process-firefox/ says:
<blockquote>
Enable / Disable Electrolysis in Firefox

To enable or disable multi-process Firefox, do the following

    Type about:config in the browser's address bar.
    Confirm that you will be careful.
    Search for browser.tabs.remote.autostart.
    Double-click on the preference.

Setting the value of browser.tabs.remote.autostart to true enables the multi-process architecture in Firefox, setting it to false disables it.

Please note that you need to restart the browser when you change the preference's value.
</blockquote>

**Comment by millwood** [comment]
It appears that firefox, at least up to 50, automatically turns electrolysis off if there are incompatible add-ons and it knows that keysnail is incompatible.

**Comment by louiz** [comment]
So, doesn’t anybody know how to replace this code

`key.generateKey(ev.originalTarget, KeyEvent.DOM_VK_DOWN, true);`

with something that works with electrolysis?

**Comment by jgeerds** [comment]
Is there any update on this?
